src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/germline/CoverageModellerGermlineSparkToggle.java

```diff
@@ -0,0 +1,363 @@
+        }
+
+        logger.info("Parsing the germline contig ploidy annotation table...");
+        final GermlinePloidyAnnotatedTargetCollection ploidyAnnotatedTargetCollection;
+        try (final Reader ploidyAnnotationsReader = getReaderFromURI(contigPloidyAnnotationsURI)) {
+            ploidyAnnotatedTargetCollection = new GermlinePloidyAnnotatedTargetCollection(ContigGermlinePloidyAnnotationTableReader
+                    .readContigGermlinePloidyAnnotationsFromReader(contigPloidyAnnotationsURI, ploidyAnnotationsReader),
+                    readCounts.targets());
+        } catch (final IOException ex) {
+            ex.printStackTrace();
+            throw new UserException.CouldNotReadInputFile("Could not parse the germline contig ploidy annotations table");
+        }
+
+        logger.info("Parsing the copy number transition prior table and initializing the caches...");
+        final IntegerCopyNumberTransitionProbabilityCacheCollection transitionProbabilityCacheCollection;
+        try (final Reader copyNumberTransitionPriorTableReader = getReaderFromURI(copyNumberTransitionPriorTableURI)) {
+            final String parentURI = new File(copyNumberTransitionPriorTableURI).getParent();
+            transitionProbabilityCacheCollection = new IntegerCopyNumberTransitionProbabilityCacheCollection(
+                    copyNumberTransitionPriorTableReader, parentURI, true);
+        } catch (final IOException ex) {
+            ex.printStackTrace();
+            throw new UserException.CouldNotReadInputFile("Could not parse the copy number transition prior table");
+        }
+        final IntegerCopyNumberExpectationsCalculator integerCopyNumberExpectationsCalculator =
+                new IntegerCopyNumberExpectationsCalculator(transitionProbabilityCacheCollection,
+                        params.getMinLearningReadCount());
+
+        final CoverageModelParameters model;
+        if (modelPath != null) {
+            logger.info("Loading model parameters...");
+            model = CoverageModelParameters.read(modelPath);
+        } else {
+            model = null;
+        }
+
+        logger.info("Initializing the EM algorithm workspace...");
+        final CoverageModelEMWorkspaceGermline workspace = new CoverageModelEMWorkspaceGermline(
+                readCounts, ploidyAnnotatedTargetCollection, sexGenotypeDataCollection,
+                integerCopyNumberExpectationsCalculator, params, model, ctx);
+        final CoverageModelEMAlgorithm<IntegerCopyNumberState> algo = new CoverageModelEMAlgorithm<>(params,
+                workspace);
+
+        switch (jobType) {
+            case LEARN_AND_CALL:
+                algo.runExpectationMaximization();
+                logger.info("Saving the model to disk...");
+                workspace.saveModel(new File(outputPath, FINAL_MODEL_SUBDIR).getAbsolutePath());
+                break;
+
+            case CALL_ONLY:
+                algo.runExpectation();
```

Comment:
For whatever reason, Github is not letting me reply to the comment on `ex.printStackTrace()` directly above this.  Looks like these instances were not cleaned up.